staging/src/k8s.io/apiserver/pkg/admission/attributes_test.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"strconv"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSetAnnotation(t *testing.T) {
+	annotations := make(map[string]string)
+	attr := &attributesRecord{annotations: annotations, annotationLock: sync.Mutex{}}
+
+	// test overwrite
+	attr.SetAnnotation("podsecuritypolicy.admission.k8s.io", "policy", "privileged")
+	attr.SetAnnotation("podsecuritypolicy.admission.k8s.io", "policy", "privileged-overwrite")
+	assert.Equal(t, annotations["podsecuritypolicy.admission.k8s.io/policy"], "privileged-overwrite", "admission annotations can be overwritten")
+
+	// test invalid plugin name
+	attr.SetAnnotation("INVALID-DNS-Subdomain", "policy", "privileged")
+	assert.Equal(t, annotations["INVALID-DNS-Subdomain/policy"], "", "invalid pluginName is not allowed ")
+
+	// test set concurrently
```

Comment:
Even if it works, it's a pretty flacky test, isn't it?

Reply:
Will this test make sort of sense, when run with `-race` option?